design/nfs.md

```diff
@@ -0,0 +1,157 @@
+
+## Why NFS?
+
+NFS is widely used for persistent storage in kubernetes cluster. Using NFS storage is a convenient and easy way to provision storage for applications.
+An NFS volume allows an existing NFS (Network File System) share to be mounted into the pod.
+The contents of an NFS volume are preserved and the volume is merely unmounted if the pod is stopped/destroyed. This means that an NFS volume can be pre-populated with data, and that data can be “handed off” between pods.
+NFS supports multiple read/write simultaneously so a single share can be attached to multiple pods.
+
+## Design
+
+Rook brings storage platforms to Kubernetes with a fully Kubernetes-native design. It integrates deeply into cloud native environments leveraging extension points and providing a seamless experience for scheduling, lifecycle management, resource management, security, monitoring, and user experience.
+With this project Rook is providing another widely adopted storage option for admins and users of cloud-native environments. This project will automate NFS starting from its configuration (such as allowed hosts, read/write permissions etc.) to deployment and provisioning. The operations on NFS which cannot be done natively by Kubernetes will be automated.
+NFS doesn’t provide an internal provisioner for kubernetes, so Rook is needed as an external provisioner.
+
+### Initial Setup
+
+The flow of creating NFS backed storage in Rook is
+1. The settings are determined and saved in an NFS server CRD (rook-nfs.yaml)
+2. `kubectl create -f rook-nfs.yaml`
+3. Rook is notified of the new CRD
+   - Rook starts the NFS daemon on the desired nodes in the cluster
+   - Rook creates a storage class
+4. The pod which requires NFS storage creates a PVC to consume the volume.
+
+### Deployed service
+
+When the NFS CRD instance is created, Rook responds to this request by starting the NFS daemon with the configuration and exports stated in the CRD and creates a service to expose NFS.
+
+### NFS CRD
+
+The NFS CRD spec will specify the following:
+1. NFS server configuration
+    - export (The directory being exported)
+    - host (The host or network to which the export is being shared)
+    - options (The options to be used for the host) e.g., read and write permission, sync, root_squash.
+2. NFS server storage backend configuration. E.g., configuration for various storage backends(ceph, ebs, azure disk etc) that will be shared using NFS.
+
+The parameters to configure NFS are:
+
+| Parameter                       | Description                                                              | Default              |
+|---------------------------------|--------------------------------------------------------------------------|----------------------|
+| `replicas`                      | The no. of NFS daemon to start                                            | `1`                  |
```

Comment:
Are there any recommendations about the number of daemons to start? For example, are there any perf or locking implications?